frontend/src/components/App/PluginSettings/PluginSettingsDetails.tsx

```diff
@@ -0,0 +1,193 @@
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import Stack from '@mui/material/Stack';
+import withStyles from '@mui/styles/withStyles';
+import _ from 'lodash';
+import { isValidElement, useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import helpers from '../../../helpers';
+import { deletePlugin } from '../../../lib/k8s/apiProxy';
+import { ConfigStore } from '../../../plugin/configStore';
+import { PluginInfo, reloadPage } from '../../../plugin/pluginsSlice';
+import { useTypedSelector } from '../../../redux/reducers/reducers';
+import NotFoundComponent from '../../404';
+import { SectionBox } from '../../common';
+import { ConfirmDialog } from '../../common/Dialog';
+import ErrorBoundary from '../../common/ErrorBoundary';
+
+export default function PluginSettingsDetails() {
+  const pluginSettings = useTypedSelector(state => state.plugins.pluginSettings);
+  const { name } = useParams<{ name: string }>();
+  const decodedName = decodeURIComponent(name);
+  const plugin = pluginSettings.find(plugin => plugin.name === decodedName);
+  if (!plugin) {
+    return <NotFoundComponent />;
```

Comment:
This is an early return when you have what looks like a hook (store.useConfig) below. This can mess up React.
If it's indeed a hook and you still want to have an early return, you have to create a new component that does those config related inits + returns PluginSettingsDetailsPure itself.